sqlglot/parser.py

```diff
@@ -1099,8 +1099,35 @@ class Parser(metaclass=_Parser):
         "WITH": lambda self: self.expression(
             exp.Properties, expressions=self._parse_wrapped_properties()
         ),
+        "BUCKET": lambda self: self._parse_partitioned_by_bucket_or_truncate(
+            exp.PartitionedByBucket
+        ),
+        "TRUNCATE": lambda self: self._parse_partitioned_by_bucket_or_truncate(
+            exp.PartitionByTruncate
+        ),
```

Comment:
Perhaps we could also check the uppercase `_prev` text here instead of passing in the type.